### ISSUE TITLE
PartialRatio not working as expected

### ISSUE THREAD

**Issue by fab1aan** [JakeBayer/FuzzySharp#8]
The lib does not work as expected if there are multiple matches. I expect the method to return the match with the highest ratio.

```
string dePN = "Partnernummer";
int ratio = Fuzz.PartialRatio(dePN, text);
```

Example of texts, which behave not as expected:
text = "Partne\nrnum\nmerASDFPartnernummerASDF"; => Ratio == 85 (should be 100)
text = "PartnerrrrnummerASDFPartnernummerASDF"; => Ratio == 77 (should be 100)

**Comment by JakeBayer** [comment]
You are correct, that is not the expected behavior. I will take a look at this when I get home from work tonight.

**Comment by fab1aan** [comment]
Are there any updates on this issue? Thanks!

**Comment by JakeBayer** [comment]
Hey, sorry I have not had much time to work on this. The issue lies within
the "GetMatchingBlocks" method I found online which does not seem to handle
cases where large blocks of repeating text occur in the source string.
Unfortunately I have not discovered a way to amend this while maintaining
the O(a*b) runtime of the method, where {a,b} are the lengths of the input
strings. To make matters worse, the code I copied uses very short variable
names with little to no comments, making the understanding of the code more
difficult.

The issue lies within the following file:
https://github.com/JakeBayer/FuzzySharp/blob/master/FuzzySharp/Levenshtein.cs
If
you have any desire to check it out yourself, this would be the place to
start.
This is merely a side project of mine, so I don't have a ton of time to
admit to this, though I should have a little more now that the first round
of the American holiday season is over. I will attempt to give this another
look when I have free time this week.

**Comment by blackholeearth** [comment]
repo owner  mentions that he ported java version to c# in readme. 

java version also has partial score issue : https://github.com/xdrop/fuzzywuzzy/issues/88
